src/test/java/org/gridsuite/study/server/EquipmentInfosServiceTests.java

```diff
@@ -155,6 +179,83 @@ public void searchEquipmentInfosWithPattern() {
         assertTrue(hits.contains(otherLineInfos));
     }
 
+    private static EquipmentInfos toEquipmentInfos(Identifiable<?> i) {
+        return EquipmentInfos.builder()
+            .networkUuid(EquipmentInfosServiceTests.NETWORK_UUID)
+            .id(i.getId())
+            .name(i.getNameOrId())
+            .type(EquipmentType.getType(i).name())
+            .voltageLevels(Set.of(VoltageLevelInfos.builder().id("vlid").name("vlname").build()))
+            .build();
+    }
+
+    @Rule
+    public ErrorCollector pbsc = new ErrorCollector();
+
+    private void testFullAscii(String pat) {
+        Set<EquipmentInfos> hits;
+
+        hits = studyService.searchEquipments(NETWORK_UUID, pat, EquipmentInfosService.FieldSelector.NAME)
+            .collect(Collectors.toSet()).block();
```

Comment:
Add a test for ID ?

Reply:
It would need to modify testCase.xiidm in a way that is more invasive and may make other (older) tests fail. Balanced with the fact that equipmentName is widely tested and changes are the same for equipmentId and equipmentName, I'd rather not try and add this redundancy.

Reply:
I added a test just for internal spaces